### PR TITLE
Python: Fix warnings from pytest

### PR DESCRIPTION
**Pull request by Fokko**
Because TestType contains test in the name, pytest tries to collect it and run tests on it. Setting `__test__ = False` tells PyTest to ignore it

Also, s3 emits an error because we clean up buckets that still has content in it, so I've added some code to clean them up after we use the bucket.

### CODE REVIEW

File: python/pyproject.toml [hunk 1]
```diff
@@ -109,6 +109,10 @@ markers = [
     "s3: marks a test as requiring access to s3 compliant storage (use with --aws-access-key-id, --aws-secret-access-key, and --endpoint-url args)",
     "adlfs: marks a test as requiring access to adlfs compliant storage (use with --adlfs.account-name, --adlfs.account-key, and --adlfs.endpoint args)"
 ]
+# Turns an warning into an error
```
Comment (ajantha-bhat) [inline python/pyproject.toml:112]: nit: 

```suggestion
# Turns a warning into an error
```

Comment (Fokko) [reply python/pyproject.toml:112]: Ah nice, thanks!

File: python/tests/catalog/test_glue.py [hunk 2]
```diff
@@ -67,7 +67,18 @@ def get_random_databases(n: int) -> Set[str]:
...
 def fixture_s3_bucket(_s3) -> None:  # type: ignore
-    _s3.create_bucket(Bucket=BUCKET_NAME)
+    bucket = _s3.create_bucket(Bucket=BUCKET_NAME)
+    yield bucket
+
+    response = _s3.list_objects_v2(
+        Bucket=BUCKET_NAME,
+    )
+    while response["KeyCount"] > 0:
+        _s3.delete_objects(Bucket=BUCKET_NAME, Delete={"Objects": [{"Key": obj["Key"]} for obj in response["Contents"]]})
```
Comment (amogh-jahagirdar) [inline python/tests/catalog/test_glue.py:77]: I haven't checked the details of this test, but delete_objects only accepts 1000 keys at a time for a given bucket. We're probably not creating close to that much here which is why this is passing, but just so the logic is robust we probably want to handle that (if not here, we can do in a separate PR as well)

Comment (amogh-jahagirdar) [inline python/tests/catalog/test_glue.py:77]: I haven't checked the details of this test, but delete_objects only accepts 1000 keys at a time for a given bucket. We're probably not creating close to that much here which is why this is passing, but just so the logic is robust we probably want to handle that (if not here, we can do in a separate PR as well)

**Review by amogh-jahagirdar — APPROVED** [review]
One comment, it's non-blocking IMO for this since I don't think it's an issue when we actually run the tests but I think we should fix it at some point just so it's more robust

File: python/tests/catalog/test_glue.py (hunk 2 above)
Comment (Fokko) [reply python/tests/catalog/test_glue.py:77]: I'm also no expert on the subject, but it looks like the max number of key returned is also 1000:
image
From: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.list_objects_v2

This will iterate until `KeyCount` is zero

File: python/tests/test_transforms.py [hunk 3]
```diff
@@ -422,6 +422,7 @@ def test_void_transform() -> None:
 
 class TestType(IcebergBaseModel):
     __root__: Transform[Any, Any]
+    __test__ = False
```
Comment (rdblue) [inline python/tests/test_transforms.py:425]: Was this fixed in the other PR?

File: python/tests/catalog/test_glue.py [hunk 4]
```diff
@@ -67,7 +67,18 @@ def get_random_databases(n: int) -> Set[str]:
 
 @pytest.fixture(name="_bucket_initialize")
 def fixture_s3_bucket(_s3) -> None:  # type: ignore
-    _s3.create_bucket(Bucket=BUCKET_NAME)
+    bucket = _s3.create_bucket(Bucket=BUCKET_NAME)
+    yield bucket
```
Comment (rdblue) [inline python/tests/catalog/test_glue.py:71]: Does this change the return type?

**Review by rdblue — APPROVED** [review]
Looks good to me when tests are passing.

File: python/tests/catalog/test_glue.py (hunk 4 above)
Comment (Fokko) [reply python/tests/catalog/test_glue.py:71]: No, this is a bit of an interesting fixture. It didn't return anything before, and now it just blocks the yield until the test is passed, and then cleans it up.

File: python/tests/test_transforms.py (hunk 3 above)
Comment (Fokko) [reply python/tests/test_transforms.py:425]: I think that was a PR to the 0.3.0 branch, so we can create another RC from that branch without having the latest pyarrow changes released along with it.

**Comment by Fokko** [comment]
This PR needs a bit of TLC; it looks like the adlfs test also throws some warnings that we need to fix.
